plugins/MauticSocialBundle/Form/Type/FacebookLoginType.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace MauticPlugin\MauticSocialBundle\Form\Type;
+
+use Mautic\CoreBundle\Factory\MauticFactory;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\ChoiceList\ChoiceList;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+
+/**
+ * Class FacebookLoginType
+ *
+ * @package Mautic\FormBundle\Form\Type
+ */
+class FacebookLoginType extends AbstractType
+{
+    /**
+     * @var MauticFactory
+     */
+    private $factory;
+
+    public function __construct(MauticFactory $factory)
+    {
+        $this->factory = $factory;
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        /** @var \Mautic\PluginBundle\Helper\IntegrationHelper $integrationHelper */
+        $integrationHelper = $this->factory->getHelper('integration');
+        $integrationObject = $integrationHelper->getIntegrationObject('Facebook');
+        $disabled = true;
+
+        if ($integrationObject->getIntegrationSettings()->isPublished()) {
+            /** @var \Mautic\AssetBundle\Model\AssetModel $model */
+            $model = $this->factory->getModel('form');
+
+            $integration = array(
+                'integration' => $integrationObject->getName(),
+            );
+
+            $builder->add(
+                'authUrl',
+                'hidden',
+                array(
+                    'data' => $model->buildUrl('mautic_integration_auth_user', $integration, true, array()),
+                )
+            );
+            $disabled = false;
+
+        }
+
+        $builder->add(
+            'width',
+            'text',
+            array(
+                'label_attr' => array('class' => 'control-label'),
+                'label' => 'mautic.integration.Facebook.login.width',
+                'disabled' => $disabled,
+                'required' => false,
+                'attr' => array(
+                    'class' => 'form-control',
+                    'placeholder' => 'mautic.integration.Facebook.login.width',
+                    'preaddon' => 'fa'
```

Comment:
Did you mean to have an empty pre-addon here? Just having fa results in just a gray box. Can we either remove this line altogether (and 93) or add a font awesome icon like `fa fa-something`. 

![](http://alan.direct/drop/2016-04-25_13-50-19.png)